File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/structure/OneToManyModelTest.java
package com.raizlabs.android.dbflow.test.structure;

import com.raizlabs.android.dbflow.annotation.OneToMany;
import com.raizlabs.android.dbflow.sql.language.Delete;
import com.raizlabs.android.dbflow.sql.language.Select;
import com.raizlabs.android.dbflow.test.FlowTestCase;

import java.util.List;

/**
 * Description: Tests the {@link OneToMany} annotation to ensure it works as expected.
 */
public class OneToManyModelTest extends FlowTestCase {

    public void testOneToManyModel() {
        Delete.tables(TestModel2.class, OneToManyModel.class);

        TestModel2 testModel2 = new TestModel2();
        testModel2.name = "Greater";
        testModel2.order = 4;
        testModel2.save();

        testModel2 = new TestModel2();
        testModel2.name = "Lesser";
        testModel2.order = 1;
        testModel2.save();

        // assert we save
        OneToManyModel oneToManyModel = new OneToManyModel();
        oneToManyModel.name = "HasOrders";
        oneToManyModel.save();
        assertTrue(oneToManyModel.exists());

        // assert loading works as expected.
        oneToManyModel = new Select().from(OneToManyModel.class).querySingle();
        assertNotNull(oneToManyModel.orders);
        assertTrue(!oneToManyModel.orders.isEmpty());

        // assert the deletion cleared the variable
        oneToManyModel.delete();
        assertFalse(oneToManyModel.exists());
        assertNull(oneToManyModel.orders);

        // assert singular relationship was deleted.
        List<TestModel2> list = new Select().from(TestModel2.class).queryList();
        assertTrue(list.size() == 1);

        Delete.tables(TestModel2.class, OneToManyModel.class);
    }

}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/models/OneToManyModelTest.kt
package com.raizlabs.android.dbflow.models

import com.raizlabs.android.dbflow.BaseUnitTest
import com.raizlabs.android.dbflow.TestDatabase
import com.raizlabs.android.dbflow.config.database
import com.raizlabs.android.dbflow.query.select
import com.raizlabs.android.dbflow.query.list
import com.raizlabs.android.dbflow.query.result
import com.raizlabs.android.dbflow.structure.delete
import com.raizlabs.android.dbflow.structure.exists
import com.raizlabs.android.dbflow.structure.save
import org.junit.Assert.*
import org.junit.Test

class OneToManyModelTest : BaseUnitTest() {

    @Test
    fun testOneToManyModel(){
        database(TestDatabase::class) {
            var testModel2 = TwoColumnModel("Greater", 4)
            testModel2.save()

            testModel2 = TwoColumnModel("Lesser", 1)
            testModel2.save()

            // assert we save
            var oneToManyModel = OneToManyModel("HasOrders")
            oneToManyModel.save()
            assertTrue(oneToManyModel.exists())

            // assert loading works as expected.
            oneToManyModel = (select from OneToManyModel::class).result!!
            assertNotNull(oneToManyModel.getRelatedOrders(this))
            assertTrue(!oneToManyModel.getRelatedOrders(this).isEmpty())

            // assert the deletion cleared the variable
            oneToManyModel.delete()
            assertFalse(oneToManyModel.exists())
            assertNull(oneToManyModel.orders)

            // assert singular relationship was deleted.
            val list = (select from TwoColumnModel::class).list
            assertTrue(list.size == 1)
        }
    }
}